test/handlers/redeem_and_unstake/UnstakeProgressedHandler/persist.test.ts

```diff
@@ -0,0 +1,102 @@
+    const save = sinon.stub();
+
+    const mockedRepository = sinon.createStubInstance(MessageRepository,
+      {
+        save: Promise.resolve(save as any),
+        get: Promise.resolve(null),
+      });
+    const handler = new UnstakeProgressedHandler(mockedRepository as any);
+
+    const models = await handler.persist(transactions);
+
+    const expectedModel = new Message(
+      transactions[0]._messageHash,
+    );
+    expectedModel.sender = transactions[0]._redeemer;
+    expectedModel.direction = MessageDirection.AuxiliaryToOrigin;
+    expectedModel.targetStatus = MessageStatus.Progressed;
+    expectedModel.type = MessageType.Redeem;
+    expectedModel.gatewayAddress = transactions[0].contractAddress;
+    expectedModel.secret = transactions[0]._unlockSecret;
+
+    assert.equal(models.length, transactions.length, 'Number of models must be equal to transactions');
+    SpyAssert.assert(mockedRepository.save, 1, [[expectedModel]]);
+    SpyAssert.assert(mockedRepository.get, 1, [[transactions[0]._messageHash]]);
+  });
+
+  it('should not change message record if current status is already progressed', async () => {
+    const save = sinon.stub();
+
+    const existingMessageWithProgressStatus = new Message(web3utils.keccak256('1'));
+    existingMessageWithProgressStatus.targetStatus = MessageStatus.Progressed;
+    const mockedRepository = sinon.createStubInstance(MessageRepository,
+      {
+        save: Promise.resolve(save as any),
+        get: Promise.resolve(existingMessageWithProgressStatus),
+      });
+    const handler = new UnstakeProgressedHandler(mockedRepository as any);
+
+    const models = await handler.persist(transactions);
+
+    const expectedModel = new Message(
+      transactions[0]._messageHash,
+    );
+    expectedModel.targetStatus = MessageStatus.Progressed;
+    expectedModel.secret = transactions[0]._unlockSecret;
+
+    assert.equal(models.length, transactions.length, 'Number of models must be equal to transactions');
+    SpyAssert.assert(mockedRepository.save, 1, [[expectedModel]]);
+    SpyAssert.assert(mockedRepository.get, 1, [[transactions[0]._messageHash]]);
+  });
+
+  it('should not change message record if current status is not Declared', async () => {
+    const save = sinon.stub();
+
+    const existingMessageWithProgressStatus = new Message(web3utils.keccak256('1'));
+    existingMessageWithProgressStatus.sourceStatus = MessageStatus.RevocationDeclared;
+    const mockedRepository = sinon.createStubInstance(MessageRepository,
+      {
+        save: Promise.resolve(save as any),
+        get: Promise.resolve(existingMessageWithProgressStatus),
+      });
+    const handler = new UnstakeProgressedHandler(mockedRepository as any);
+
+    const models = await handler.persist(transactions);
+
+    const expectedModel = new Message(
+      transactions[0]._messageHash,
+    );
+    expectedModel.sourceStatus = existingMessageWithProgressStatus.sourceStatus;
```

Comment:
I also missed similar thing in RedeemIntentConfirmed handler PR, please fix it there to.